amethyst_network/src/server/udp.rs

```diff
@@ -0,0 +1,125 @@
+//! All UDP related logic for getting and sending data out to the other side.
+
+use std::net::SocketAddr;
+use std::sync::mpsc::{self, Receiver, Sender};
+use std::thread;
+
+use super::{
+    ClientEvent, PacketReceiving, PacketSending, ReceiveHandler, SendHandler, ServerSocketEvent,
+};
+use laminar::net::UdpSocket;
+use laminar::NetworkConfig;
+
+/// An UDP receiver, wrapper for starting the UDP-receiving thread.
+pub struct UdpReceiver {
+    // socket used for receiving packets
+    socket: UdpSocket,
+}
+
+impl UdpReceiver {
+    /// Create a new instance of UDP server.
+    pub fn new(addr: SocketAddr) -> UdpReceiver {
+        let socket = UdpSocket::bind(&addr, NetworkConfig::default()).unwrap();
+        UdpReceiver { socket }
+    }
+
+    /// This will run the udp receiver on it's own thread.
+    pub fn run(addr: SocketAddr) -> ReceiveHandler {
+        let mut udp_receiver = UdpReceiver::new(addr);
+
+        // channel used for communicating about received packets.
+        let (tx, rx) = mpsc::channel();
+
+        let thread_handle = thread::spawn(move || {
+            udp_receiver.start_receiving(tx);
+        });
+
+        ReceiveHandler::new(5000, rx, thread_handle)
```

Comment:
where does this 5000 come from?

Reply:
Don't know actually, and I'm not sure at how many packets we should break the receiving. This would be eventually done with congestion prevention.

Reply:
I will make it work via the `ServerConfig`

Reply:
👍 we should add this to some sort of config.